gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCalls.java

```diff
@@ -115,21 +115,26 @@ public boolean setException(Throwable throwable) {
 
   private static class FutureListener<T> extends HttpJsonClientCall.Listener<T> {
     private final HttpJsonFuture<T> future;
+    private T message;
+    private boolean isMessageReceived;
 
     private FutureListener(HttpJsonFuture<T> future) {
       this.future = future;
+      this.isMessageReceived = false;
     }
 
     @Override
     public void onMessage(T message) {
-      if (!future.set(message)) {
+      if (isMessageReceived) {
```

Comment:
Looks like `isMessageReceived` and `message` are always set together, can we just check if `message` is null?

Reply:
I think you're right. I wasn't sure if parameter `message` would ever be passed in as null, but it seems like that won't be the case. I think unknown fields will be ignored and null values will be set to their default values and any invalid JSON would have an exception throw so we should be good. I'll make the change

Reply:
I gave it more thought and I think we should revert to your previous approach. The logic is fine in `onMessage`, but in `onClose`, we can not use `if(storedMessage == null)` to determine anything because we cannot distinguish if the response itself is null or we haven't received response yet.

Reply:
Sounds good. I can revert to the previous way.